docs/training-run/index.txt

```diff
@@ -4,106 +4,255 @@
-suggestions on data loading.
+   python3 -m determined.launch.torch_distributed --rdzv_endpoint=$CUSTOM_RDZV_ADDR -- --trial model_def:MyTrial
 
-Since project directories might include large artifacts that should not be packaged as part of the
-model definition (e.g., data sets or compiled binaries), users can optionally include a
-``.detignore`` file at the top level that specifies file paths to be omitted from the model
-definition. The ``.detignore`` file uses the same syntax as `.gitignore
-<https://git-scm.com/docs/gitignore>`__. Note that byte-compiled Python files (e.g., ``.pyc`` files
-or ``__pycache__`` directories) are always ignored.
+DeepSpeed Launcher
+==================
 
-Local Test Mode
+Format:
+
+``determined.launch.deepspeed [[DEEPSPEED_ARGS...] --] (--trial TRIAL)|(SCRIPT...)``
+
+The DeepSpeed launcher launches a training script under deepspeed with automatic handling of IP
+addresses, node and container communication, and fault tolerance. See the official DeepSpeed
+documentation for information about how to use the DeepSpeed launcher.
+
+Example:
+
+.. code:: bash
+
+   python3 -m determined.launch.deepspeed --trial model_def:MyTrial
+
+Legacy Launcher
 ===============
 
-The local test mode is to sanity-check your training code and run a compressed version of the full
-experiment circle. It can help you debug the errors in your code without running the full experiment
-circle that is expensive in the resources it takes and sometimes needs a long time to get ready. It
-helps with quick iteration on the code. You can run the following command:
+Format:
+
+``entrypoint: model_def:TrialClass``
+
+The entrypoint field expects a predefined or custom script, but also supports legacy file and trial
+class definitions.
+
+When you specify a trial class as the entry point, it must be a subclass of a Determined trial
+class.
+
+Each trial class is designed to support one deep learning application framework. When training or
+validating models, the trial might need to load data from an external source so the training code
+needs to define data loaders.
+
+A TrialClass is located in the ``model_def`` filepath and launched automatically. This is considered
+legacy behavior. By default, this configuration automatically detects distributed training, based on
+slot size and number of machines, and launches with Horovod for distributed training. If used in a
+distributed training context, the entrypoint is:
+
+.. code:: bash
+
+   python3 -m determined.launch.horovod --trial model_def:TrialClass
+
+Nested Launchers
+================
+
+Entrypoint supports nesting multiple launch layers in a single script. This can be useful for tasks
+that need to be run before the training code starts, such as profiling tools (dlprof), custom memory
+management tools (numactl), or data preprocessing.
+
+Example:
+
+.. code:: bash
+
+   dlprof --mode=simple python3 -m determined.launch.autohorovod --trial model_def:MnistTrial
```

Comment:
question for perhaps @liamcli or @coreystatendet or @garrett361, since I have no idea and couldn't find the answer online... Will this command line work?  There will be several processes between the dlprof main process and the process that actually does training.  I'm concerned that dlprof isn't actually going to work with those extra processes (determined.launch.horovod and horovodrun).

Separately, the name is no longer `autohorovod` but just `horovod`

Reply:
@rb-determined-ai i've tested dlprof for pytorch before. it depends on the level of profiling you want to do. `dlprof --mode=simple` works as a launcher for broad GPU usage metrics. for something more granular (ie --mode=pytorch), you need to implement some hooks in the pytorch training code to tell dlprof where the markers are. but this is just an example of a launch layer, so i think that level of detail belongs elsewhere

Reply:
Left it as-is.